src/middlewared/middlewared/plugins/cron.py

```diff
@@ -195,3 +198,68 @@ def cron_extend(self, data):
         await self.middleware.call('service.restart', 'cron')
 
         return response
+
+    @accepts(
+        Int('id')
+    )
+    @job()
+    async def run(self, job, id):
+        """
+        Job to run cronjob task of `id`.
+        """
+        cron_task = await self._get_instance(id)
+        cron_cmd = ' '.join(
+            (await self.construct_cron_command(
+                cron_task['schedule'], cron_task['user'],
+                cron_task['command'], cron_task['stdout'], cron_task['stderr']
+            ))[6:]
+        )
+
+        job.set_progress(
+            10,
+            'Executing Cron Task'
+        )
+
+        p = await Popen(
+            cron_cmd, stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT, shell=True
+        )
+        stdout, stderr = await p.communicate()
+
+        job.set_progress(
+            85,
+            'Executed Cron Task'
+        )
+
+        if stdout:
+            email = (await self.middleware.call('user.query', [['username', '=', 'root']], {'get': True}))['email']
+            stdout = stdout.decode()
+            if email:
+                mail_job = await self.middleware.call(
+                    'mail.send', {
+                        'subject': 'CronTask Manual Run',
+                        'text': stdout,
+                        'to': [email]
+                    }
+                )
+
+                job.set_progress(
+                    95,
+                    'Sending mail for Cron Task output'
+                )
+
+                await mail_job.wait()
+                if mail_job.error:
+                    self.middleware.logger.debug(f'Failed to send email for CronTask manual run: {mail_job.error}')
+            else:
+                job.set_progress(
+                    95,
+                    'Email for root user not configured. Skipping sending mail.'
+                )
+
+            self.middleware.logger.debug(f'Executed CronTask - {cron_cmd}: {stdout}')
```

Comment:
This is a job, if we want we can put logs to it, using logs_excerpt, lets not pollute middleware log with this